cpp/tests/ast/transform_tests.cpp

```diff
@@ -343,6 +343,71 @@ TEST_F(TransformTest, ImbalancedTreeArithmeticDeep)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view(), verbosity);
 }
 
+TEST_F(TransformTest, DeeplyNestedArithmeticLogicalExpression)
+{
+  // Test logic for deeply nested arithmetic and logical expressions.
+  constexpr int64_t left_depth_level  = 100;
+  constexpr int64_t right_depth_level = 75;
+
+  auto generate_ast_expr = [](int64_t depth_level,
+                              cudf::ast::column_reference col_ref,
+                              cudf::ast::ast_operator root_operator,
+                              cudf::ast::ast_operator arithmetic_operator,
+                              bool nested_left_tree) {
+    std::vector<cudf::ast::operation> expressions;
```

Comment:
Please use a `std::list` here -- the reason is that AST expressions may hold _references_ to other AST expressions, and putting these into a `std::vector` can invalidate those references if the vector is reallocated due to changing size.

Perhaps we should copy the comment from here:
https://github.com/rapidsai/cudf/blob/fcc89503c1f1e15ec287519959013adcf2bf8a52/cpp/benchmarks/ast/transform.cpp#L65-L68

```suggestion
    // Note that a std::list is required here because of its guarantees against reference invalidation
    // when items are added or removed. References to items in a std::vector are not safe if the
    // vector must re-allocate.
    auto expressions = std::list<cudf::ast::operation>();
```

Reply:
I totally forgot about this. Thanks for the catch Bradley. This was my bad, I suggested the change in an earlier round of iteration.